gk/fib.c

```diff
@@ -60,31 +65,91 @@ gk_lpm_add_ipv6_route(uint8_t *ip,
  * callers must ensure that the entry is not being used.
  */
 static inline int
-initialize_ether_cache(struct ether_cache *eth_cache)
+clear_ether_cache(struct ether_cache *eth_cache)
 {
-	memset(eth_cache, 0, sizeof(*eth_cache));
+	memset(eth_cache->fields_to_clear, 0, sizeof(*eth_cache) -
+		offsetof(struct ether_cache, fields_to_clear));
 
 	return 0;
 }
 
-/*
- * Fill up the Ethernet cached header.
- * Note that the destination MAC address should be filled up by LLS.
- */
-static inline void
-fill_up_ether_cache_locked(struct ether_cache *eth_cache,
-	struct ipaddr *addr, struct gatekeeper_if *iface)
+static void
+gk_arp_and_nd_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
 {
+	struct ether_cache *eth_cache = arg;
+
+	if (eth_cache->ref_cnt == 0) {
```

Comment:
The test should be `pcall_again == NULL`, if true, call `clear_ether_cache()` and return.